### DISCUSSION TITLE
ConfigAndCredentialsCacheManager.cpp:172: Aws::String Aws::Config::GetCachedConfigValue(const String&, const String&): Assertion `s_configManager' failed. Aborted (core dumped)

### DISCUSSION THREAD

**Discussion by dimu637** [aws/aws-sdk-cpp#2605] [Q&A] [answered]
I try to run a rocksdbcloud example but I get
```
ConfigAndCredentialsCacheManager.cpp:172: Aws::String Aws::Config::GetCachedConfigValue(const String&, const String&): Assertion `s_configManager' failed.
Aborted (core dumped)
```
Is there any configuration I missed?
Thx

**Comment by yasminetalby** [comment]
Hello @dimu637 ,

Thank you very much for reaching out.
Unfortunately, there is not enough information for me to provide more support here. 
Could you provide more information of your use case? rocksdbcloud is not an AWS owned library.  
You need to either reach out to their team or provide the example used as well as any modification/environment set up steps you have followed.

Thank you very much for your time and collaboration.

Sincerely,

Yasmine

**Comment by dimu637** [comment]
Thanks for the response
I try to run the example they provide here
https://github.com/rockset/rocksdb-cloud/blob/master/cloud/examples/cloud_durable_example.cc
and get the error message 
```
~/rocksdb-cloud/cloud/examples$ ./cloud_durable_example
cloud_durable_example: ConfigAndCredentialsCacheManager.cpp:172: Aws::String Aws::Config::GetCachedConfigValue(const String&, const String&): Assertion `s_configManager' failed.
Aborted (core dumped)
```
Could you please take a look at their example code? I am not sure whether they use it in the wrong way, or it is a bug in was SDK, or I missing some configurations.

Thx.

**Comment by yasminetalby** [comment]
Hello @dimu637 ,

Thank you very much for linking the example. Could you please also provide all configuration and step up steps followed from your end as well as:
- AWS CPP SDK version used
- Compiler and Version used
- Operating System and version

In general, I would recommend to first reach out to the owner of the repository for guidance on their work. If it turns out that this is a bug from the SDK (or a misuse from their end) they will then open an issue with us directly. 

Best,

Yasmine

**Reply by dimu637** [reply]
AWS CPP SDK version used: the main branch
```
:~$ gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
:~$ cmake --version
cmake version 3.16.3

Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.6 LTS
Release:	20.04
```
I try to reach out to them but no response yet.

**Comment by sem32** [comment, marked as answer]
hey @dimu637 
I have had the same issue
try to call Aws::InitAPI() before Aws::Client::ClientConfiguration;
